clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -59,200 +83,65 @@ public void testSerdeFromUnknown() {
         }
-        deserializer.close();
     }
 
-    @Test
+    @Test(expected = SerializationException.class)
     public void floatDeserializerShouldThrowSerializationExceptionOnTooFewBytes() {
-        final Deserializer<Float> deserializer = Serdes.Float().deserializer();
-        try {
-            deserializer.deserialize(topic, new byte[3]);
-            fail("Should have thrown a SerializationException because of too few input bytes");
-        } catch (SerializationException e) {
-            // Ignore (there's no contract on the details of the exception)
+        try (Serde<Float> serde = Serdes.Float()) {
+            serde.deserializer().deserialize(topic, new byte[3]);
         }
-        deserializer.close();
     }
 
 
-    @Test
+    @Test(expected = SerializationException.class)
     public void floatDeserializerShouldThrowSerializationExceptionOnTooManyBytes() {
-        final Deserializer<Float> deserializer = Serdes.Float().deserializer();
-        try {
-            deserializer.deserialize(topic, new byte[5]);
-            fail("Should have thrown a SerializationException because of too many input bytes");
-        } catch (SerializationException e) {
-            // Ignore (there's no contract on the details of the exception)
+        try (Serde<Float> serde = Serdes.Float()) {
+            serde.deserializer().deserialize(topic, new byte[5]);
         }
-        deserializer.close();
     }
 
     @Test
     public void floatSerdeShouldPreserveNaNValues() {
-        final int someNaNAsIntBits = 0x7f800001;
-        final float someNaN = Float.intBitsToFloat(someNaNAsIntBits);
-        final int anotherNaNAsIntBits = 0x7f800002;
-        final float anotherNaN = Float.intBitsToFloat(anotherNaNAsIntBits);
-
-        final Serde<Float> serde = Serdes.Float();
-        // Because of NaN semantics we must assert based on the raw int bits.
-        final Float roundtrip = serde.deserializer().deserialize(topic,
-            serde.serializer().serialize(topic, someNaN));
-        assertThat(Float.floatToRawIntBits(roundtrip), equalTo(someNaNAsIntBits));
-        final Float otherRoundtrip = serde.deserializer().deserialize(topic,
-            serde.serializer().serialize(topic, anotherNaN));
-        assertThat(Float.floatToRawIntBits(otherRoundtrip), equalTo(anotherNaNAsIntBits));
-
-        serde.close();
-    }
-
-    @Test
-    public void testDoubleSerializer() {
-        Double[] doubles = new Double[]{
-            5678567.12312d,
-            -5678567.12341d
-        };
-
-        Serializer<Double> serializer = Serdes.Double().serializer();
-        Deserializer<Double> deserializer = Serdes.Double().deserializer();
-
-        for (Double value : doubles) {
-            assertEquals("Should get the original double after serialization and deserialization",
-                    value, deserializer.deserialize(topic, serializer.serialize(topic, value)));
+        int someNaNAsIntBits = 0x7f800001;
```

Comment:
I know it should but all vars like these in all test methods (not just in this class) are not final. Just for that...